controllers/market/updater.js

```diff
@@ -0,0 +1,157 @@
+const { Market, Warehouse, Bid, PlayerFantasy, Op } = require('../../databases/db');
+const responses = require('../../utils/responses/responses');
+
+const bidUpdate = async (req, res) => {
+    const userId = req.user.id.id;
+    const eventId = req.eventId;
+    const bidId  = req.params.bidId;
+    let { value, marketId, isDirectPurchase = false } = req.body;
+    value = parseInt(value);
+    if (value == null || marketId == null) {
+        return responses.errorDTOResponse(res, 400, 'Debe proporcionar un valor y un marketId');
+    }
+
+    const market = await Market.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [{
+                eventId: eventId
+            }, {
+                id: marketId
+            }]
+        }
+    });
+    const bid = await Bid.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [{
+                userId: userId
+            },{
+                marketId: marketId
+            }, {
+                id: bidId
+            }]
+        }
+    });
+    const player = await PlayerFantasy.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [
+            {
+                userId: userId
+            }, {
+                eventId: eventId
+            }]
+        }
+    })
+
+    if (!bid) {
+        return responses.errorDTOResponse(res, 400, 'No existe esta oferta');
+    }
+
+    if (!player) {
+        return responses.errorDTOResponse(res, 400, 'No estas participando en este evento');
+    }
+    if (!market) {
+        return responses.errorDTOResponse(res, 400, 'No existe esta subasta');
+    }
+    if (player.money < value) {
```

Comment:
Creo que esto es inncesario, cuando es compra directa podemos omitir el value pt4

Aaaah..... con razon lo hiciste asi, bueno:
```suggestion
    if ((!isDirectPurchase && player.money < value) || (isDirectPurchase && player.money < market.immediatePurchaseValue)) {
```
Con derivaciones booleanas esto se puede optimizar, pero bueno, te lo dejo a ti o a @DaikoneKisu esa optimizacion